test/unit/select_plan_test.lua

```diff
@@ -339,3 +339,203 @@ g.test_table_count = function()
     t.assert_equals(utils.table_count(
         {['Ivan'] = 1, ['Peter'] = 2, ['Fyodor'] = 3, ['Alexander'] = 4}), 4)
 end
+
+local get_sharding_key_from_scan_value_cases = {
+    get_sharding_key_from_scan_value_for_non_table_value = {
+        -- Input values.
+        scan_value = 2,
+        scan_index = 'id',
+        scan_iter = box.index.EQ,
+        sharding_index = 'id',
+        -- Expected values.
+        sharding_key = 2,
+    },
+    get_sharding_key_from_scan_value_for_empty_value = {
+        -- Input values.
+        scan_value = nil,
+        scan_index = 'id',
+        scan_iter = box.index.EQ,
+        sharding_index = 'id',
+        -- Expected values.
+        sharding_key = nil,
+    },
+    get_sharding_key_from_scan_value_for_ge_iter_returns_nil = {
+        -- Input values.
+        scan_value = 2,
+        scan_index = 'id',
+        scan_iter = box.index.GE,
+        sharding_index = 'id',
+        -- Expected values.
+        sharding_key = nil,
+    },
+    get_sharding_key_from_scan_value_returns_nil_if_sharding_index_is_not_scan_index = {
+        -- Input values.
+        scan_value = 2,
+        scan_index = 'id',
+        scan_iter = box.index.EQ,
+        sharding_index = 'age',
+        -- Expected values.
+        sharding_key = nil,
+    },
+    get_sharding_key_from_scan_value_for_table_value = {
+        -- Input values.
+        scan_value = { 'John', 'Doe' },
+        scan_index = 'age',
+        scan_iter = box.index.EQ,
+        sharding_index = 'age',
+        -- Expected values.
+        sharding_key = { 'John', 'Doe' },
+    },
+    get_sharding_key_from_scan_value_for_partial_table_value_returns_nil = {
+        -- Input values.
+        scan_value = { nil, 'Doe' },
+        scan_index = 'age',
+        scan_iter = box.index.EQ,
+        sharding_index = 'age',
+        -- Expected values.
+        sharding_key = nil,
+    },
+}
+
+for name, case in pairs(get_sharding_key_from_scan_value_cases) do
+    g[('test_%s'):format(name)] = function()
+        local scan_index = box.space.customers.index[case.scan_index]
+        local sharding_index = box.space.customers.index[case.sharding_index]
+
+        local sharding_key = select_plan.internal.get_sharding_key_from_scan_value(case.scan_value, scan_index,
```

Comment:
Personally for me long names looks awful. I would make a shortcut for it.
```lua
local get_key = select_plan.internal.get_sharding_key_from_scan_value
local sharding_key = get_key(case.scan_value, scan_index,
                             case.scan_iter, sharding_index)
t.assert_equals(sharding_key, case.sharding_key)
```

Feel free to ignore a comment and close.

Reply:
Used a shortcut